File: Projects/_PrestaSoft/PrestaSoft/frontend/SoftBibliotecaWA/SoftBibliotecaPrestamosBO/PrestamoBO.cs
using SoftBibliotecaBO;
using SoftBibliotecaBO.ServicioWeb;

using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace SoftBibliotecaPrestamosBO
{
    public class PrestamoBO : BaseBO
    {
        public prestamo obtenerPorId(int idPrestamo)
        {
            return this.WsCliente.obtenerPorId_Prestamo(idPrestamo);
        }
        

        public BindingList<prestamo> mostrarPrestamosEnCurso(int idUsuario, int limite)
        {
            try
            {
                prestamo[] prestamos = this.WsCliente.mostrarPrestamosEnCurso(idUsuario, limite);
                if (prestamos == null) return new BindingList<prestamo>();
                return new BindingList<prestamo>(prestamos);
            }
            catch (Exception ex)
            {
                return new BindingList<prestamo>();
            }
        }

        public BindingList<prestamo> mostrarUltimosPrestamos(int limite)
        {
            try
            {
                prestamo[] prestamos = this.WsCliente.mostrarUltimosPrestamos(limite);
                if (prestamos == null) return new BindingList<prestamo>();
                return new BindingList<prestamo>(prestamos);
            }
            catch (Exception ex)
            {
                return new BindingList<prestamo>();
            }
        }

        public int solicitarAmpliacion(int idPrestamo)
        {
            return this.WsCliente.solicitarAmpliacion(idPrestamo);
        }

        public int confirmarAmpliacion(int idPrestamo)
        {
            return this.WsCliente.confirmarAmpliacion(idPrestamo);
        }

        public BindingList<string> consultarCalificacionPrestamo(int idUsuario, int idCopiaEjemplar)
        {
            try
            {
                string[] prestamos = this.WsCliente.consultarCalificacionPrestamo(idUsuario, idCopiaEjemplar);
                if (prestamos == null) return new BindingList<string>();
                return new BindingList<string>(prestamos);
            }
            catch (Exception ex)
            {
                return new BindingList<string>();
            }
        }

        public int registrarPrestamo(int idUsuario, int idCopiaEjemplar)
        {
            return this.WsCliente.registrarPrestamo(idUsuario, idCopiaEjemplar);
        }


        public prestamo consultarPrestamoQueSePrestoCopiaEjemplar(int idCopiaEjemplar)
        {
            return this.WsCliente.consultarPrestamoQueSePrestoCopiaEjemplar(idCopiaEjemplar);
        }

        public int confirmarDevolucion(int idPrestamo, int idCopiaEjemplar, int estaEnMalEstado)
        {
            return this.WsCliente.confirmarDevolucion(idPrestamo, idCopiaEjemplar, estaEnMalEstado);
        }
        //int idUsuario, string fechaMin, string fechaMax, string estadoDePrestamos, int idUniversidad, int mostrarONoReservas, string descripcion
        public byte[] reporteSolicitudes(int idUsuario, string fechaMin, string fechaMax, string estadoDePrestamos, int idUniversidad, int mostrarONoReservas, string descripcion)
        {
            return this.WsReporte.reporteSolicitudes(idUsuario,  fechaMin, fechaMax, estadoDePrestamos, idUniversidad, mostrarONoReservas, descripcion); 
        }

    }
}

File: Projects/_PrestaSoft/PrestaSoft/backend/ServidorRMI/SoftBibliotecaRMIServidor/src/pe/edu/pucp/softbiblioteca/rmi/interfaces/PrestamoBO.java

package pe.edu.pucp.softbiblioteca.rmi.interfaces;
import java.rmi.Remote;
import java.rmi.RemoteException;
import java.util.ArrayList;
import pe.edu.pucp.softbiblioteca.prestamos.model.Prestamo;

public interface PrestamoBO extends Remote{
    
    public Integer insertar(String instantePrestamo,String fechaDevolucionEsperada, String fechaDevolucionReal,
                            String tipoEstado,Integer huboReservaPrevia,Integer idEjemplarAsociado,
                            Integer idUsuarioAsociado) throws RemoteException;
    
    public Integer modificar(Integer idPrestamo,String instantePrestamo,String fechaDevolucionEsperada, String fechaDevolucionReal,
                            String tipoEstado,Integer huboReservaPrevia,Integer idEjemplarAsociado,
                            Integer idUsuarioAsociado) throws RemoteException;
    
    public Integer eliminar(Integer idPrestamo) throws RemoteException;
    
    public ArrayList<Prestamo> listarTodos(String instantePrestamo,String fechaDevolucionEsperada, String fechaDevolucionReal,String tipoEstado,
                                           Integer huboReservaPrevia,Integer idEjemplarAsociado,Integer idUsuarioAsociado, Integer limite) throws RemoteException;
    
    public Prestamo obtenerPorId(Integer idPrestamo) throws RemoteException;
       
    public ArrayList<Prestamo> mostrarPrestamosEnCurso(Integer idUsuario, Integer limite) throws RemoteException;
    
    public ArrayList<Prestamo> mostrarUltimosPrestamos(Integer limite) throws RemoteException;
    
    public Integer solicitarAmpliacion(Integer idPrestamo) throws RemoteException;
    
    public Integer confirmarAmpliacion(Integer idPrestamo) throws RemoteException;
    
    public ArrayList<String> consultarCalificacionPrestamo(Integer idUsuario, Integer idCopiaEjemplar) throws RemoteException;

    public Integer registrarPrestamo(Integer idUsuario, Integer idCopiaEjemplar) throws RemoteException;

    public Prestamo consultarPrestamoQueSePrestoCopiaEjemplar(Integer idCopiaEjemplar) throws RemoteException;
    
    public int confirmarDevolucion(Integer idPrestamo,Integer  idCopiaEjemplar, Integer estaEnMalEstado) throws RemoteException;
    
}
